@GeorgeGayno-NOAA 
I was pushed from a wrong folder. Now it is corrected.
But still have one cannot do it correctly.
enc_jpeg2000.c:79: warning: Member MAXOPTSSIZE (macro definition) of file enc_jpeg2000.c is not documented.
This is a macro definition, but I used the way that suggested by doxygen mannuel. It still there.

--- a/src/dec_jpeg2000.c
+++ b/src/dec_jpeg2000.c
@@ -41,7 +41,7 @@
  * @param[in] injpc Input JPEG2000 code stream.
  * @param[in] bufsize Length (in bytes) of the input JPEG2000 code stream.
  * @param[in] outfld Output matrix of grayscale image values. Only grayscale is expected.
- * @return 0 = Successful decode
+ * @return - 0 = Successful decode
  *         - -3 = Error decode jpeg2000 code stream.
  *         - -5 = decoded image had multiple color components.
  *

--- a/src/enc_jpeg2000.c
+++ b/src/enc_jpeg2000.c
@@ -56,9 +56,10 @@
  * 1 = try increasing number of guard bits otherwise, no additional options
  * @param[in] jpclen  - Number of bytes allocated for new JPEG2000 code stream in outjpc.
  * @param[in] outjpc - Output encoded JPEG2000 code stream
- * @return 0 = Length in bytes of encoded JPEG2000 code stream
- *         -  -3 = Error decode jpeg2000 code stream.
- *         -  -5 = decoded image had multiple color components.
+ * @return 
+ * - 0 Length in bytes of encoded JPEG2000 code stream.
+ * - -3 Error decode jpeg2000 code stream.
+ * - -5 decoded image had multiple color components.
  * Only grayscale is expected.
  *
  * @note Requires JasPer Software version 1.500.4 or 1.700.2
@@ -73,10 +74,7 @@ int SUB_NAME(unsigned char *cin,g2int *pwidth,g2int *pheight,g2int *pnbits,
     jas_image_t image;
     jas_stream_t *jpcstream,*istream;
     jas_image_cmpt_t cmpt,*pcmpt;
-/**
- * \def MAXOPTSSIZE
-*/
-#define MAXOPTSSIZE 1024 
+#define MAXOPTSSIZE 1024 /**< Maximum size of the options. */
     char opts[MAXOPTSSIZE];
 
     g2int width,height,nbits;

--- a/src/gridtemplates.f
+++ b/src/gridtemplates.f
@@ -44,7 +44,8 @@
 
       module gridtemplates
 
-      integer,parameter :: MAXLEN=200,MAXTEMP=31
+      integer,parameter :: MAXLEN=200 !< maximum number of octets in mapgrid
+      integer,parameter :: MAXTEMP=31 !< maximum number of entries in the template
 
       type gridtemplate
           integer :: template_num
@@ -53,7 +54,7 @@
           logical :: needext
       end type gridtemplate
 
-      type(gridtemplate),dimension(MAXTEMP) :: templates
+      type(gridtemplate),dimension(MAXTEMP) :: templates !< a gridtemplate type
 
       data templates(1)%template_num /0/     !  Lat/Lon 
       data templates(1)%mapgridlen /19/

